src/components/SlidingNavigation/SlidingNavigation.js

```diff
@@ -0,0 +1,112 @@
+import * as React from "react";
+import PropTypes from "prop-types";
+import classNames from "classnames";
+
+import "./SlidingNavigation.css";
+
+const SlidingNavigation = ({ data, options }) => {
+  const activeItemClassName = `SlidingNavigation__item--active`;
+  const observerOptions = options || {
+    threshold: 0.2,
+  };
+
+  React.useEffect(() => {
+    const observer = new IntersectionObserver((entries) => {
+      entries.forEach((entry) => {
+        if (
+          entry.isIntersecting &&
+          !window.location.hash.includes(entry.target.id)
+        ) {
+          window.history.replaceState({}, ``, `#${entry.target.id}`);
+          const currentActiveElement = document.querySelector(
+            `.${activeItemClassName}`
+          );
+
+          // Remove current active className
+          if (currentActiveElement) {
+            currentActiveElement.classList.remove(activeItemClassName);
+          }
+
+          const nextActiveElement = document.getElementById(
+            `${entry.target.id}-sn`
+          );
+
+          // Add an active className
+          if (nextActiveElement) {
+            nextActiveElement.classList.add(activeItemClassName);
+          }
+
+          // Scroll to current active navigation item
+          // in the SlidingNavigation
+          const navigationWrapper =
+            document.querySelector(`.SlidingNavigation`);
+          if (navigationWrapper) {
+            navigationWrapper.scrollTo({
+              top: 0,
+              left:
+                nextActiveElement.offsetLeft -
+                nextActiveElement.offsetWidth / 2,
+              behavior: `smooth`,
+            });
+          }
+        }
+      });
+    }, observerOptions);
+
+    data.forEach((item) => {
+      observer.observe(document.getElementById(item.linkId));
+    });
+  }, []);
+
+  return (
+    <div className="SlidingNavigation">
+      {data.map((item) => {
+        return (
+          <div
+            id={`${item.linkId}-sn`}
+            className={classNames(`SlidingNavigation__item`, {
+              "SlidingNavigation__item--active": window.location.hash.includes(
+                item.linkId
+              ),
+            })}
+            key={item.linkId}
+          >
+            <div
+              onClick={() => {
+                const element = document.getElementById(item.linkId);
+                // @TODO: need to make dynamic values and not hardcoded numbers
+                window.scrollTo({
```

Comment:
https://www.npmjs.com/package/react-anchor-link-smooth-scroll try this.

Reply:
Will try, looks interesting. My main concern is if it also supports synchronous scrolls, since there are two scrolls happening, because for example `scrollIntoView()` cancels out each other

Reply:
Looking more at it, this seems like my first implementation with `<Link to={nodeID} />` except that it scrolls instead of jumping to it.
I could probably still use this for this specific navigation item, but is it worth it to install a library for that? The `window.scrollTo()` method is widely supported by browsers

Reply:
> https://www.npmjs.com/package/react-anchor-link-smooth-scroll try this.

Do not polyfill, there's no need to push even more JS to the user for such a small feature. It's native browser functionality at this point.